backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -874,8 +874,17 @@ object Compiler {
       case _ => false
     }
     val updatedClasspath =
-      if (needsRtJar) inputs.classpath ++ RtJarCache.create(JavaRuntime.version, logger)
-      else inputs.classpath
+      if (needsRtJar) {
+        val possibleRtJar = inputs.javacBin
+          .map { binary =>
+            binary.getParent.resolve("../jre/lib/rt.jar")
+          }
+          .filter(_.exists)
+        possibleRtJar match {
+          case Some(rtJar) => inputs.classpath :+ rtJar
+          case None => inputs.classpath ++ RtJarCache.create(JavaRuntime.version, logger)
```

Comment:
```suggestion
        inputs.classpath ++ possibleRtJar.orElse(RtJarCache.create(JavaRuntime.version, logger))
```

Reply:
I would even rewrite the entire thing to:
```scala
val possibleRtJar = 
  if (needsRtJar) 
    inputs.javacBin
      .map(_.resolve("../../jre/lib/rt.jar"))
      .filter(_.exists)
      .orElse(RtJarCache.create(JavaRuntime.version, logger))
  else None
val updatedClasspath = inputs.classpath ++ possibleRtJar
```

Reply:
Done!